homeassistant/components/fritz/sensor.py

```diff
@@ -282,8 +295,8 @@ async def async_setup_entry(
     ):
         pass
 
-    for sensor_type, sensor_data in SENSOR_DATA.items():
-        if not dsl and sensor_data.get("connection_type") == DSL_CONNECTION:
+    for sensor_type in SENSOR_TYPES:
+        if not dsl and sensor_type.connection_type == DSL_CONNECTION:
             continue
         entities.append(FritzBoxSensor(fritzbox_tools, entry.title, sensor_type))
```

Comment:
You could use a list comprehension here as well. The initial `entities = []` would then be redundant. Furthermore, the `if entities` below is also not necessary.